src/signing.rs

```diff
@@ -512,23 +592,227 @@ impl TryFrom<pkcs8::PrivateKeyInfo<'_>> for Keypair {
 }
 
 #[cfg(feature = "serde")]
-impl Serialize for Keypair {
+impl Serialize for SigningKey {
     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
     where
         S: Serializer,
     {
-        let bytes = &self.to_bytes()[..];
-        SerdeBytes::new(bytes).serialize(serializer)
+        SerdeBytes::new(&self.secret_key).serialize(serializer)
     }
 }
 
 #[cfg(feature = "serde")]
-impl<'d> Deserialize<'d> for Keypair {
+impl<'d> Deserialize<'d> for SigningKey {
     fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
     where
         D: Deserializer<'d>,
     {
         let bytes = <SerdeByteBuf>::deserialize(deserializer)?;
-        Keypair::from_bytes(bytes.as_ref()).map_err(SerdeError::custom)
+        Self::try_from(bytes.as_ref()).map_err(SerdeError::custom)
```

Comment:
Just for recordkeeping, this is the diff I got from the previous version of this block of code in `signing.rs`:
```diff
0a1
>
23c24
< // bitcoind, a user might wish to have one master keypair from which others are
---
> // bitcoind, a user might wish to have one master signing key from which others are
51c52
< impl<'a> From<&'a SecretKey> for ExpandedSecretKey {
---
> impl From<&SecretKey> for ExpandedSecretKey {
67c68
<     fn from(secret_key: &'a SecretKey) -> ExpandedSecretKey {
---
>     fn from(secret_key: &SecretKey) -> ExpandedSecretKey {
73c74
<         h.update(secret_key.as_bytes());
---
>         h.update(secret_key);
93c94
<     pub(crate) fn sign(&self, message: &[u8], public_key: &PublicKey) -> ed25519::Signature {
---
>     pub(crate) fn sign(&self, message: &[u8], verifying_key: &VerifyingKey) -> ed25519::Signature {
104c105
<         R = (&r * &constants::ED25519_BASEPOINT_TABLE).compress();
---
>         R = (&r * &ED25519_BASEPOINT_TABLE).compress();
108c109
<         h.update(public_key.as_bytes());
---
>         h.update(verifying_key.as_bytes());
125c126
<     /// * `public_key` is a [`PublicKey`] which corresponds to this secret key.
---
>     /// * `verifying_key` is a [`VerifyingKey`] which corresponds to this secret key.
141c142
<         public_key: &PublicKey,
---
>         verifying_key: &VerifyingKey,
188c189
<         R = (&r * &constants::ED25519_BASEPOINT_TABLE).compress();
---
>         R = (&r * &ED25519_BASEPOINT_TABLE).compress();
196c197
<             .chain_update(public_key.as_bytes())
---
>             .chain_update(verifying_key.as_bytes())
203d203
<     }
```

Reply:
Aah sorry, some of the comments might have some find/replace changes which don't make sense.

I tried to look for as many of those as I could find but it seems like there are a few I missed.

Reply:
Reverted "master signing key" => "master keypair" in d6b0876